Move database and config from homeassistant /config to addon /config

felipecrs opened the pull request:
## Proposed change
<!--
  Describe what this pull request does and how it will benefit users of Frigate.
  Please describe in detail any considerations, breaking changes, etc. that are 
  made in this pull request.
-->

Adds a migration to move the config and database from Home Assistant's config dir to the addon's config dir for when Frigate is running as a Home Assistant add-on.

Closes #16334
Sibling of https://github.com/blakeblackshear/frigate-hass-addons/pull/210

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code
- [ ] Documentation Update

## Checklist

<!--
  Put an `x` in the boxes that apply.
-->

- [ ] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] The code has been formatted using Ruff (`ruff format frigate`)

felipecrs committed:
Move database and config from homeassistant /config to addon /config

felipecrs committed:
Merge branch 'dev' of https://github.com/blakeblackshear/frigate into addon_config

felipecrs committed:
Merge branch 'dev' of https://github.com/blakeblackshear/frigate into addon_config

felipecrs committed:
Re-implement config migration for the add-on

felipecrs committed:
Align some terms

felipecrs committed:
Improve function name

felipecrs committed:
Use local variables

felipecrs committed:
Add model.path migration

felipecrs committed:
Fix homeassistant config path

felipecrs committed:
Ensure migration scripts run before go2rtc and frigate

felipecrs committed:
Migrate all files I know

felipecrs committed:
Merge branch 'dev' into addon_config

felipecrs committed:
Add ffmpeg.path migration

hawkeye217 commented on docker/main/rootfs/etc/s6-overlay/s6-rc.d/prepare/run:
```diff
@@ -0,0 +1,138 @@
+        if [[ -d "${old_ffmpeg_path}" ]]; then
+            local new_ffmpeg_dir
+            new_ffmpeg_dir="$(dirname "${ffmpeg_path}")"
+            echo "[INFO] Migrating model from '${old_ffmpeg_path}' to '${ffmpeg_path}'..."
```
```suggestion
            echo "[INFO] Migrating ffmpeg from '${old_ffmpeg_path}' to '${ffmpeg_path}'..."
```

felipecrs replied:
Nice catch! :D

felipecrs committed:
Update docker/main/rootfs/etc/s6-overlay/s6-rc.d/prepare/run

Co-authored-by: Josh Hawkins <32435876+hawkeye217@users.noreply.github.com>

felipecrs committed:
Improve some variable names and organization

felipecrs committed:
Update docs to reflect addon config dir

felipecrs committed:
Merge branch 'dev' of https://github.com/blakeblackshear/frigate into addon_config

felipecrs commented:
@NickM-27, @hawkeye217 thanks for reviewing this PR.

I did not change anything on it since your last review, except I updated the documentation.

Mind taking a look on the docs changes too?

felipecrs committed:
Update live.md with /addon_configs

felipecrs commented:
Actually, allow me some more time. I want to do a more deep scan.

felipecrs committed:
Move addon config section to configuration doc

felipecrs commented:
I'm done with the changes to the documentation.

felipecrs commented on docs/docs/configuration/live.md:
````diff
@@ -104,9 +104,9 @@ cameras:
+- For internal/local access, unless you are running through the HassOS addon, you will also need to set the WebRTC candidates list in the go2rtc config. For example, if `192.168.1.10` is the local IP of the device running Frigate:
 
-  ```yaml title="/config/frigate.yaml"
+  ```yaml title="/addon_configs/ccab4aaf_frigate/config.yaml"
````
Yes. Maybe I should have said only config.yaml.

felipecrs commented on docs/docs/configuration/live.md:
```diff
@@ -104,9 +104,9 @@ cameras:
 
 - For external access, over the internet, setup your router to forward port `8555` to port `8555` on the Frigate device, for both TCP and UDP.
-- For internal/local access, unless you are running through the add-on, you will also need to set the WebRTC candidates list in the go2rtc config. For example, if `192.168.1.10` is the local IP of the device running Frigate:
+- For internal/local access, unless you are running through the HassOS addon, you will also need to set the WebRTC candidates list in the go2rtc config. For example, if `192.168.1.10` is the local IP of the device running Frigate:
```
Alright. I will change all places. But I cannot do it now, probably tomorrow morning. Sorry.

felipecrs committed:
Align several terminologies and improve text

felipecrs committed:
Fix webrtc example config title

felipecrs commented:
@hawkeye217, I got too excited and aligned the terms across the whole documentation. I believe you'll appreciate all these changes, but let me know if not.

felipecrs committed:
Capitalize Add-on in more places

hawkeye217 commented on docs/docs/configuration/index.md:
```diff
@@ -3,9 +3,11 @@ id: index
 ---
 
-For Home Assistant Addon installations, the config file needs to be in the root of your Home Assistant config directory (same location as `configuration.yaml`). It can be named `frigate.yaml` or `frigate.yml`, but if both files exist `frigate.yaml` will be preferred and `frigate.yml` will be ignored.
+For Home Assistant Add-on installations, the config file should be at `/addon_configs/ccab4aaf_frigate/config.yaml` ([click here to learn more on how to access this directory](#accessing-add-on-config)).
```
Might want to change this or clarify the directory name for the add-on variants (plus anywhere else you use this in your changes).

felipecrs replied:
This is clarified in the "click here" link. You think it's not enough?

hawkeye217 replied:
Maybe I'd change the directory name in the example to `/addon_configs/<addon_directory>/config.yaml` and then say something like "where `<addon_directory>` is specific to the version of the Add-on you are running. See the list of directories [here](#accessing-add-on-config)" or something similar.

felipecrs replied:
Sounds good. Thank you.

felipecrs replied:
@hawkeye217, made the changes. Please let me know if you liked it. :)

felipecrs committed:
Improve specific add-on config dir docs

hawkeye217 commented on docs/docs/troubleshooting/edgetpu.md:
```diff
@@ -32,7 +32,7 @@ The USB coral can draw up to 900mA and this can be too much for some on-device U
 
 - When running Frigate in a VM, Proxmox lxc, etc. you must ensure both device IDs are mapped.
-- When running HA OS you may need to run the Full Access version of the Frigate addon with the `Protected Mode` switch disabled so that the coral can be accessed.
+- When running trough the Home Assistant OS you may need to run the Full Access variant of the Frigate Add-on with the _Protection mode_ switch disabled so that the coral can be accessed.
```
```suggestion
- When running through the Home Assistant OS you may need to run the Full Access variant of the Frigate Add-on with the _Protection mode_ switch disabled so that the coral can be accessed.
```

felipecrs replied:
Oops, my bad.

hawkeye217 commented on docs/docs/configuration/index.md:
```diff
@@ -3,9 +3,11 @@ id: index
-For all other installation types, the config file should be mapped to `/config/config.yml` inside the container.
+For all other installation types, the config file should be mapped to `/config/config.yaml` inside the container.
+
+It can be named `config.yaml` or `config.yml`, but if both files exist `config.yaml` will be preferred and `config.yml` will be ignored.
```
> t can be named `config.yaml` or `config.yml`, but if both files exist `config.yaml` will be preferred and `config.yml` will be ignored.

Is this still true? @NickM-27

felipecrs replied:
From what I remember from the scripts, yeah. I'm not home now, but for example you can check the go2rtc create_config.py.

felipecrs replied:
Here:

https://github.com/felipecrs/frigate/blob/a0c8cc9562ebf5ebeeee389a79824809adf9ff4b/docker/main/rootfs/usr/local/go2rtc/create_config.py#L35-L37

yaml is preferred.

NickM-27 replied:
ah, this is actually no longer true, .yml is preferred. We should update the go2rtc create_config.py to use this function as well

https://github.com/blakeblackshear/frigate/blob/e36fe797828e695520d58faccf4dd7397deabaf3/frigate/util/config.py#L20

felipecrs replied:
Should I also drop support for loading `config.json`? Or users can still set CONFIG_FILE to a json file?

hawkeye217 replied:
My vote for that would be a yes, since writing back to the config from the UI is not supported with a json config. Let's see what @NickM-27 says.

NickM-27 replied:
I believe changing the env variable to json will work, best to leave it the same for now

hawkeye217 commented on docs/docs/configuration/index.md:
```diff
@@ -23,9 +25,25 @@ cameras:
+
+**Whenever you see `/config` in the documentation, it refers to this directory.**
+
+If for example you are running the standard Add-on variant and use the [VS Code Add-on](https://github.com/hassio-addons/addon-vscode) to browse your files, you can click _File_ > _Open folder..._ and navigate to `/addon_configs/ccab4aaf_frigate` to access the Frigate `/config` directory and edit the `config.yaml` file. You can also use the built-in file editor in the Frigate UI to edit the confi ...
```
I think we should be consistent throughout the docs and use `.yml` everywhere.

hawkeye217 commented on docs/docs/configuration/live.md:
````diff
@@ -104,9 +104,9 @@ cameras:
+- For internal/local access, unless you are running through the HA Add-on, you will also need to set the WebRTC candidates list in the go2rtc config. For example, if `192.168.1.10` is the local IP of the device running Frigate:
 
-  ```yaml title="/config/frigate.yaml"
+  ```yaml title="config.yaml"
````
`.yml`

hawkeye217 commented on docs/docs/frigate/installation.md:
```diff
@@ -275,24 +274,28 @@ See [the network storage guide](/guides/ha_network_storage.md) for instructions
+
+If you are using hardware acceleration for ffmpeg, you **may** need to use the _Full Access_ variant of the Add-on. This is because the Frigate Add-on runs in a container with limited access to the host system. The _Full Access_ variant allows you to disable _Protection mode_ and give Frigate full access to the host system.
+
+You can also edit the Frigate configuration file through the [VS Code Add-on](https://github.com/hassio-addons/addon-vscode) or similar. In that case, the configuration file will be at `/addon_configs/<addon_directory>/config.yaml`, where `<addon_directory>` is specific to the variant of the Frigate Add-on you are running. See the list of directories [here](../configuration/index.md#accessing-add ...
```
`.yml`

NickM-27 commented on docs/docs/integrations/plus.md:
```diff
@@ -19,11 +19,11 @@ Once logged in, you can generate an API key for Frigate in Settings.
 ### Set your API key
 
-In Frigate, you can use an environment variable or a docker secret named `PLUS_API_KEY` to enable the `Frigate+` buttons on the Explore page. Home Assistant Addon users can set it under Settings > Addons > Frigate NVR > Configuration > Options (be sure to toggle the "Show unused optional configuration options" switch).
+In Frigate, you can use an environment variable or a docker secret named `PLUS_API_KEY` to enable the `Frigate+` buttons on the Explore page. Home Assistant Addon users can set it under Settings > Add-ons > Frigate > Configuration > Options (be sure to toggle the "Show unused optional configuration options" switch).
```
I am pretty sure it is still called `Frigate NVR` in the addon list, no?

felipecrs replied:
No, it was changed to just Frigate.

felipecrs replied:
Long time ago:

- https://github.com/blakeblackshear/frigate-hass-addons/pull/76

felipecrs committed:
Align bash and python scripts to prefer config.yml over config.yaml

felipecrs committed:
Support config.json in migration shell scripts

felipecrs committed:
Change docs to reflect config.yml is preferred over config.yaml

felipecrs committed:
If previous config was yaml, migrate to yaml

felipecrs committed:
Fix typo in edgetpu.md

felipecrs committed:
Fix formatting of Python files

felipecrs committed:
Remove HailoRT Beta add-on variant from docs

felipecrs committed:
Add migration for labelmap and certs

felipecrs commented:
Let me test it again since the last round of changes.

Feel free to review the last commits though.

felipecrs committed:
Fix variable name

felipecrs committed:
Fix new_config_file var unset

felipecrs committed:
Fix addon config directories table

felipecrs committed:
Improve db migration to avoid migrating files like .db.bak

felipecrs committed:
Fix echo location

felipecrs commented:
All good. My logs, out of curiosity:

```
s6-rc: info: service prepare: starting
[INFO] Starting migration from Home Assistant config dir to Add-on config dir...
[INFO] Migrating database from '/homeassistant/frigate.db' to '/config' dir...
copied '/homeassistant/frigate.db' -> '/config/frigate.db'
removed '/homeassistant/frigate.db'
copied '/homeassistant/frigate.db-shm' -> '/config/frigate.db-shm'
removed '/homeassistant/frigate.db-shm'
copied '/homeassistant/frigate.db-wal' -> '/config/frigate.db-wal'
removed '/homeassistant/frigate.db-wal'
[INFO] Migrating .model.path from '/homeassistant/frigate/yolo_nas_s.onnx' to '/config/frigate/yolo_nas_s.onnx'...
mkdir: created directory '/config/frigate'
copied '/homeassistant/frigate/yolo_nas_s.onnx' -> '/config/frigate/yolo_nas_s.onnx'
removed '/homeassistant/frigate/yolo_nas_s.onnx'
[INFO] Migrating '/homeassistant/model_cache' to '/config'...
[INFO] Migrating other files from '/homeassistant' to '/config'...
copied '/homeassistant/.exports' -> '/config/.exports'
removed '/homeassistant/.exports'
copied '/homeassistant/.jwt_secret' -> '/config/.jwt_secret'
removed '/homeassistant/.jwt_secret'
copied '/homeassistant/.timeline' -> '/config/.timeline'
removed '/homeassistant/.timeline'
copied '/homeassistant/.vacuum' -> '/config/.vacuum'
removed '/homeassistant/.vacuum'
copied '/homeassistant/go2rtc' -> '/config/go2rtc'
removed '/homeassistant/go2rtc'
[INFO] Migrating config file from '/homeassistant/frigate.yaml' to '/config/config.yaml'...
copied '/homeassistant/frigate.yaml' -> '/config/config.yaml'
removed '/homeassistant/frigate.yaml'
[INFO] Migration from Home Assistant config dir to Add-on config dir completed.
s6-rc: info: service prepare successfully started
```

hawkeye217 reviewed (approved):
Looks good. Thanks for the contribution!
